extensions/amp-selector/0.2/selector.js

```diff
@@ -0,0 +1,127 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as CSS from './selector.css';
+import * as Preact from '../../../src/preact';
+import {useState} from '../../../src/preact';
+
+const SelectorContext = Preact.createContext({});
+
+/**
+ * @param {!JsonObject} props
+ * @return {PreactDef.Renderable}
+ */
+export function Selector(props) {
+  const {
+    'value': value,
+    'defaultValue': defaultValue,
+    'disabled': disabled,
+  } = props;
+  const isMultiple = props['multiple'] !== undefined;
+  const [selectedState, setSelectedState] = useState(
+    value ? [].concat(value) : defaultValue ? [].concat(defaultValue) : []
```

Comment:
I think this has some issues. We had this controlled/uncontrolled case in training. Did you happen to note the solution we ended up building then?

Reply:
My notes have that controlled inputs read the `value` prop and uncontrolled inputs read the `defaultValue` prop, with the latter being preferable: "The pattern of specifying if an input is controlled means that the `event.target` is different from the React rendered output. `defaultValue` is `readonly` and only takes information from user interaction." 

However it looks like the [React docs](https://reactjs.org/docs/uncontrolled-components.html) recommend controlled inputs in the case of forms. This looks like an issue to be resolved with form integrations IMO.

Reply:
IMHO, when in doubt we should move with controlled - it's the least code and less confusion. Let's rip out `defaultValue` and related logic. We'll bring it back with a separate pull request later.